includes/class-sensei-learner.php

```diff
@@ -9,13 +9,70 @@
  * @since 1.9.0
  */
 class Sensei_Learner {
+	/**
+	 * Instance of singleton.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
 	/**
 	 * Cache of the learner terms.
 	 *
 	 * @var WP_Term[]
 	 */
 	private static $learner_terms = [];
 
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Course_Enrolment_Manager constructor. Private so it can only be initialized internally.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Sets the actions.
+	 *
+	 * @since 3.0.0
+	 */
+	public function init() {
+		// Delete user activity and enrolment terms when user is deleted.
+		add_action( 'deleted_user', array( $this, 'delete_user_registers' ) );
+	}
+
+	/**
+	 * Delete all user activity and enrolment terms when user is deleted.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param  integer $user_id User ID.
+	 * @return void
+	 */
+	public function delete_user_registers( $user_id = 0 ) {
+		if ( $user_id ) {
+			// Remove activities.
+			Sensei_Utils::delete_all_user_activity( $user_id );
+
+			// Remove enrolment terms.
+			$learner_term = self::get_learner_term( $user_id );
+			wp_delete_term( $learner_term->term_id, Sensei_PostTypes::LEARNER_TAXONOMY_NAME );
```

Comment:
Jake changed the enrolments to be saved in user metadata so we should probably update this.

Reply:
Actually, it's still saving terms here: https://github.com/Automattic/sensei/blob/master/includes/class-sensei-learner.php#L32
And the new user meta is being removed automatically with the user.
Does it make sense?

Reply:
Yup, we're still keeping the term. We're just storing the results object in the user meta.

Reply:
Cool! Sorry I got confused there.